democracy/importing/json_importer.py

```diff
@@ -175,25 +166,34 @@ def import_hearing(hearing_datum, force=False, patch=False):
     )
     assert not hearing.geojson or isinstance(hearing.geojson, dict)
     hearing.save(no_modified_at_update=True)
+    # if patching, soft delete old data to prevent duplicates
+    if patch:
+        clean_hearing_for_patching(hearing)
     main_section = hearing.sections.create(
         type=SectionType.objects.get(identifier=InitialSectionType.MAIN),
         title="",
         abstract=(hearing_datum.pop("lead") or ""),
         content=(hearing_datum.pop("body") or ""),
     )
-    import_comments(main_section, hearing_datum.pop("comments", ()), patch)
-    import_images(main_section, hearing_datum, patch)
-    import_sections(hearing, hearing_datum, force, patch)
+    import_comments(main_section, hearing_datum.pop("comments", ()))
+    import_images(main_section, hearing_datum)
+    import_sections(hearing, hearing_datum, force)
+    compact_section_ordering(hearing)
+    if hearing_datum.keys():  # pragma: no cover
+        log.warn("These keys were not handled while importing %s: %s", hearing, hearing_datum.keys())
+    return hearing
 
-    # Compact section ordering...
+
+def clean_hearing_for_patching(hearing):
```

Comment:
Yep. The lines were removed as actually they no longer are related to the Hearing in question, so trying to clean them would result in an error.